crates/types/src/rpc/transactions/l1_handler_transaction.rs

```diff
@@ -0,0 +1,128 @@
+            "L1 handler transaction only supports version 0"
+        );
+
+        // No fee on L2 for L1 handler transaction.
+        let fee = FieldElement::ZERO;
+
+        compute_hash_on_elements(&[
+            PREFIX_L1_HANDLER,
+            self.version.into(),
+            self.contract_address.into(),
+            self.entry_point_selector.into(),
+            compute_hash_on_elements(
+                &self
+                    .calldata
+                    .iter()
+                    .map(|felt| FieldElement::from(*felt))
+                    .collect::<Vec<FieldElement>>(),
+            ),
+            fee,
+            chain_id.into(),
+            self.nonce.into(),
+        ])
+        .into()
+    }
+
+    /// Creates a blockifier version of `L1HandlerTransaction`.
+    pub fn create_blockifier_transaction(&self) -> DevnetResult<BlockifierL1HandlerTransaction> {
+        let version: Felt = 0_u128.into();
+
+        let transaction = BlockifierL1HandlerTransaction {
+            tx: ApiL1HandlerTransaction {
+                contract_address: ApiContractAddress::try_from(self.contract_address)?,
+                entry_point_selector: ApiEntryPointSelector(self.entry_point_selector.into()),
+                calldata: ApiCalldata(Arc::new(self.calldata.iter().map(|f| f.into()).collect())),
+                nonce: ApiNonce(self.nonce.into()),
+                version: ApiTransactionVersion(version.into()),
+            },
+            paid_fee_on_l1: ApiFee(self.paid_fee_on_l1),
+            tx_hash: ApiTransactionHash(self.transaction_hash.into()),
+        };
+
+        Ok(transaction)
+    }
+}
+
+// TODO: for this version I didn't impl `HashProducer`
+// as the chain_id is not present in the transaction itself.
+// Should we keep the chain_id somewhere in the devnet `L1HandlerTransaction`?
+// Or we don't want HashProducder on `L1HandlerTransaction` and the `new` is fine?
+impl HashProducer for L1HandlerTransaction {
+    type Error = Error;
+
+    fn generate_hash(&self) -> DevnetResult<Felt> {
+        assert_eq!(
+            self.version,
+            FieldElement::ZERO.into(),
+            "L1 handler transaction only supports version 0"
+        );
+
+        // No fee on L2 for L1 handler transaction.
+        let fee = FieldElement::ZERO;
+
+        Ok(compute_hash_on_elements(&[
+            PREFIX_L1_HANDLER,
+            self.version.into(),
+            self.contract_address.into(),
+            self.entry_point_selector.into(),
+            compute_hash_on_elements(
+                &self
+                    .calldata
+                    .iter()
+                    .map(|felt| FieldElement::from(*felt))
+                    .collect::<Vec<FieldElement>>(),
+            ),
+            fee,
+            // TODO: How to get the chain_id at this point?
+            // Or should this be computed in an other fashion?
+            // chain_id.into(),
+            0_u128.into(),
```

Comment:
This might be problematic... I'll take a closer look tomorrow.

Reply:
Can we just add chain_id in L1HandlerTransaction and mark it as serde skip?
```
    #[serde(skip)]
    pub chain_id: u128,
```

@marioiordanov @FabijanC

Reply:
Awaiting the final answer, this implementation has been removed. Please don't hesitate after a decision is made if I need to put it back again. :+1: